simple_https_server/src/lib.rs

```diff
@@ -0,0 +1,180 @@
+Content-Length: 88\r\nContent-Type: text/html\r\nConnection: Closed\r\n\r\n<html>\r\n<body>\r\n<h1>Hello from localhost!</h1>\r\n</body>\r\n</html>\r\n\r\n";
+
+pub fn tls13server_aux<Stream>(stream: Stream, host: &str) -> Result<(), AppError>
+where
+    Stream: Read + Write,
+{
+    let mut stream = RecordStream::new(stream);
+
+    let ch_rec = stream.read_record()?;
+
+    let db = {
+        let sni = ByteSeq::from_public_slice(&host.as_bytes());
+
+        ServerDB(
+            sni,
+            Bytes::from_public_slice(&ECDSA_P256_SHA256_CERT),
+            SignatureKey::from_public_slice(&ECDSA_P256_SHA256_Key),
+            None,
+        )
+    };
+
+    let ent_s = {
+        let mut entropy = [0 as u8; 64];
+        thread_rng().fill(&mut entropy);
+
+        Entropy::from_public_slice(&entropy)
+    };
+
+    match server_accept(default_algs, db, &ch_rec, ent_s) {
+        Err(x) => {
+            println!("ServerInit Error {}", x);
+            Err(PARSE_FAILED.into())
+        }
+        Ok((sh, sf, sstate)) => {
+            println!("Negotiation Complete");
+            stream.write_record(sh)?;
+            let ccs_rec = ByteSeq::from_hex("140303000101");
+            stream.write_record(ccs_rec)?;
+            stream.write_record(sf)?;
+            //println!("Server0.5 Complete");
+
+            let ccs = stream.read_record()?;
+            check_ccs_message(ccs.declassify().native_slice())?;
+            //println!("Got CCS message");
+            let cf_rec = stream.read_record()?;
+            //println!("Got fin record");
+            let sstate = server_read_handshake(&cf_rec, sstate)?;
+            println!("Handshake Complete");
+
+            let app_rec = stream.read_record()?;
+            let (_, sstate) = server_read(&app_rec, sstate)?;
+            println!("Got HTTP Request");
+
+            let http_get_resp = ByteSeq::from_public_slice(response.as_bytes());
+            let (ap, _sstate) = server_write(app_data(http_get_resp), sstate)?;
+            stream.write_record(ap)?;
+            println!("Sent HTTP Response: Hello from localhost");
+
+            Ok(())
+        }
+    }
+}
+
+pub fn tls13server(host: &str, port: &str) -> Result<(), TLSError> {
+    let addr = ["127.0.0.1", port].join(":");
+
+    let listener = TcpListener::bind(addr).unwrap();
+
+    for stream in listener.incoming() {
+        let mut stream = stream.unwrap();
+
+        let d = Duration::new(15, 0);
+        stream
+            .set_read_timeout(Some(d))
+            .expect("set_read_timeout call failed");
+        println!("New connection established!");
+        match tls13server_aux(&mut stream, host) {
+            Ok(_) => {
+                println!("Connection complete!")
+            }
+            Err(_) => {
+                println!("Connection failed!")
+            }
+        }
+    }
+    Ok(())
+}
+
+fn check_ccs_message(buf: &[u8]) -> Result<(), TLSError> {
```

Comment:
This file really needs some cleanup. But let's do that in followups.

Reply:
https://github.com/cryspen/bertie/issues/50